src/main/java/duke/storage/StorageManager.java

```diff
@@ -0,0 +1,130 @@
+package duke.storage;
+
+import duke.Duke;
+import duke.exception.DukeInvalidAddTaskException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class StorageManager {
+    private static final String TASK_TODO = "T";
+    private static final String TASK_DEADLINE = "D";
+    private static final String TASK_EVENT = "E";
+    
+    private ArrayList<Task> tasks;
+    
+    private String filePath;
+    
+    public StorageManager(String filePath) {
+        this.filePath = filePath;
+    }
+    
+    private Task getTask(String input) {
+        String[] words = input.split("--");
+        String taskType = words[0];
+        String markDoneCharacter = words[1];
+        Task task = null;
+        
+        try {
+            switch (taskType) {
+            case TASK_TODO:
+                task = new Todo(words[2]);
+                break;
+            case TASK_DEADLINE:
+                task = new Deadline(words[2] + " /by " + words[3]);
+                break;
+            case TASK_EVENT:
+                task = new Event(words[2] + " /at " + words[3]);
+                break;
+            default:
+                Duke.getUi().printInvalidTaskInFileMessage();
+                break;
+            }
+        } catch (DukeInvalidAddTaskException | ArrayIndexOutOfBoundsException e) {
```

Comment:
do catch individual exceptions separately; this one doesn't tell which error occurred definitively